src/node_snapshotable.cc

```diff
@@ -91,6 +92,23 @@ const SnapshotData* NodeMainInstance::GetEmbeddedSnapshotData() {
   return ss.str();
 }
 
+std::unique_ptr<ExternalReferenceRegistry> SnapshotBuilder::registry_ = nullptr;
+Mutex SnapshotBuilder::snapshot_data_mutex_;
+
+const std::vector<intptr_t>& SnapshotBuilder::CollectExternalReferences() {
+  Mutex::ScopedLock lock(snapshot_data_mutex_);
+  if (registry_ == nullptr) {
+    registry_.reset(new ExternalReferenceRegistry());
+  }
+  return registry_->external_references();
+}
```

Comment:
Is this the same as this?

```suggestion
const std::vector<intptr_t>& SnapshotBuilder::CollectExternalReferences() {
  static auto registry = std::make_unique<ExternalReferenceRegistry>();
  return registry_->external_references();
}
```

Reply:
From what I could tell, probably yes (assuming there is no way to create a worker without the main instance calling it first).